### PR TITLE
bug 1361764 - pytest: the final chapter [no rush]

### PR DESCRIPTION
**Pull request by willkg**
This updates all the remaining files in `socorro/unittest/` to pytest.

### CODE REVIEW

File: socorro/unittest/external/postgresql/test_base.py [hunk 1]
```diff
@@ -135,11 +139,9 @@ def test_utc(self):
...
         sql = 'SHOW TIMEZONE'
         results = base.query(sql)
-        ok_(
-            'UTC' in results[0],
-            """Please set PostgreSQL to use the UTC timezone.
-               Documentation on how to do this is included in
-               the INSTALL instructions."""
+        assert 'UTC' in results[0], (
+            'Please set PostgreSQL to use the UTC timezone. Documentation on how to do this is '
+            'included in the installation instructions.'
```
Comment (willkg) [inline socorro/unittest/external/postgresql/test_base.py]: I kept the assertion comment here. I find it pretty curious, though. What happens if postgresql isn't using UTC? Do other tests fail?

File: socorro/unittest/processor/test_breakpad_pipe_to_json.py [hunk 2]
```diff
@@ -251,172 +248,156 @@ class TestCase(TestCase):
...
+        assert crashing_thread == 1
 
     def test_extract_module_info(self):
         info = ['Module', 'firefox.exe', '24.0.0.4925', 'firefox.pdb',
                 '9FFDDF56AADE45988C759EF5ABAE53862', '0x00400000',
                 '0x004e0fff', '1']
         d = DotDict()
         bpj._extract_module_info(info, d, 17)
-        ok_('modules' in d)
-        ok_(len(d.modules), 1)
```
Comment (willkg) [inline socorro/unittest/processor/test_breakpad_pipe_to_json.py]: This is one of a bunch of `ok_` that probably should have been `eq_`.

File: socorro/unittest/external/postgresql/test_base.py [hunk 3]
```diff
@@ -93,7 +91,13 @@ def setUp(self):
...
 
         cursor = self.connection.cursor()
-
+        # NOTE(willkg): Sometimes, there are items in the reports and this causes test_counts to
+        # fail. This truncates it first. It'd be nice to fix whatever is causing this problem,
+        # though.
+        cursor.execute("""
+            TRUNCATE reports CASCADE;
+        """)
+        self.connection.commit()
```
Comment (willkg) [inline socorro/unittest/external/postgresql/test_base.py]: This test fails for me periodically and it's irritating. I hit failures when I was working on this PR. Figured I'd just throw this in here and a note.

File: socorro/unittest/processor/test_breakpad_pipe_to_json.py (hunk 2 above)
Comment (lonnen) [reply socorro/unittest/processor/test_breakpad_pipe_to_json.py]: rgr. I think I saw this above too

File: socorro/unittest/processor/test_symbol_cache_manager.py [hunk 4]
```diff
@@ -181,54 +168,29 @@ def test_process_IN_MODIFY_2(self):
...
-            "g1"
-        )
-        # unknown size suffix
-        self.assertRaises(
-            ValueError,
-            from_string_to_parse_size,
-            "1g"
-        )
+        with pytest.raises(ValueError):
+            from_string_to_parse_size(17)
```
Comment (lonnen) [inline socorro/unittest/processor/test_symbol_cache_manager.py]: +1

Comment (lonnen) [reply socorro/unittest/processor/test_symbol_cache_manager.py]: this showed up oddly in the diff, but is ok

File: socorro/unittest/cron/jobs/test_matviews.py [hunk 5]
```diff
@@ -156,15 +154,9 @@ def mock_utc_now():
                              'build-adu-matview',
                              'graphics-device-matview',):
 
-                ok_(app_name in information, app_name)
```
[the remark this answers is not in the document]
Comment (willkg) [reply socorro/unittest/cron/jobs/test_matviews.py]: `ok_` is just shorthand for:

```
def ok_(test, msg=None):
    if msg:
        assert test, msg
    else:
        assert test
```

or something along those lines.

In the non-pytest world, if this was false and the second argument wasn't there, you'd get the stunningly unhelpful:

```
AssertionError
```

So this has a second argument to indicate which `app_name` failed the assertion.

In pytest, if there's an assertion error, pytest walks the stack and tells you all about what's going on at the point of failure. So I've been nixing the msg arguments.

**Comment by willkg** [comment]
🎉 

Thank you so much for doing this grueling review work!

**Pull request merged**